src/lib/fragments/AgGrid.react.js

```diff
@@ -1042,6 +1049,38 @@ export default class DashAgGrid extends Component {
         }
     }
 
+    scrollTo(reset = false) {
+        const {gridApi} = this.state;
+        const {scrollTo, setProps} = this.props;
+        if (!gridApi) {
+            return;
+        }
+        const rowPosition = scrollTo.rowPosition ? scrollTo.rowPosition : 'top';
+        if (scrollTo.rowIndex || scrollTo.rowIndex === 0) {
+            gridApi.ensureIndexVisible(scrollTo.rowIndex, rowPosition);
+        } else if (scrollTo.rowId) {
+            const node = gridApi.getRowNode(scrollTo.rowId);
+            gridApi.ensureNodeVisible(node, rowPosition);
+        } else if (scrollTo.data) {
+            gridApi.forEachNode((node) => {
+                if (includes(node.data, [scrollTo.data])) {
```

Comment:
```suggestion
                if (equals(node.data, scrollTo.data)) {
```
But if there's a `getRowId` we can do better, right?
```
const node = gridApi.getRowNode(getRowId(scrollTo.data));
gridApi.ensureNodeVisible(node, rowPosition);
```

Reply:
To use `getRowId` we would have to always specify the function in the prop. It would be very similar to just using `rowId`

Reply:
getRowId exists, and we use it in the setSelection for the selectedRows. This is without defining it.

Reply:
Also, equals will not work in this instance. I believe that node.data has a few extra keys compared to the regular data from the df.

Reply:
@Farkites has already taken the suggestion, but @BSd3v `equals` is what was happening here anyway with `includes`. From https://ramdajs.com/docs/#includes:
> Returns `true` if the specified value is equal, in [`R.equals`](https://ramdajs.com/docs/#equals) terms, to at least one element of the given list; `false` otherwise.

So my suggestion was an exact replacement (and apparently works, or the tests would have failed!)

Reply:
We can leave it as is for now, but I can still see a purpose using `data` when there's a `getRowId` defined, and accelerating the lookup with the ID. Maybe you know which fields are involved in the ID but the function to put them together is awkward - especially since you're likely constructing it in Python but the function is normally executed in JS where there could be subtle differences.

Reply:
I think we can still use the getRowId, beats going through an entire set of data if we can just jump to the node.